Add new predefined block identifiers ``safe`` and ``finalized``.

fselmo opened the pull request:
### What was wrong?

Related to #2641

### How was it fixed?

- Added new block identifiers `safe` and `finalized` as options. `xfail`ed tests for now until we migrate our test suite to PoS.
- For filters, added appropriate documentation stating that, for `eth_newFilter`, the new block identifiers are not yet in the specifications so they may or may not yield expected results across nodes.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fpbs.twimg.com%2Fmedia%2FDoxa4c5U0AACgRC.jpg%3Alarge&f=1&nofb=1)

pacrob reviewed (approved):
lgtm

kclowes reviewed (commented):
I'm not sure if you want feedback on this right at the moment, but it looks like there is a predefined block number test [here](https://github.com/ethereum/web3.py/blob/master/tests/core/utilities/test_is_predefined_block_number.py#L18) that could use the new predefined block names. I also dug in a little bit more, and I'm pretty sure that we do want to accept these block names to the filter method. Looks good otherwise!

fselmo commented:
> I'm not sure if you want feedback on this right at the moment, but it looks like there is a predefined block number test [here](https://github.com/ethereum/web3.py/blob/master/tests/core/utilities/test_is_predefined_block_number.py#L18) that could use the new predefined block names. I also dug in a little bit more, and I'm pretty sure that we do want to accept these block names to the filter method. Looks good otherwise!

Yeah it looks like event filters may or may not take `safe` and `finalized` in the `fromBlock` and `toBlock`. Since they aren't yet in the specifications, I added support for them but I updated the documentation with notes that they may or may not yield expected results.

wolovim reviewed (approved):
looks sane to me, based on current support for these filters in clients 👍
